From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 30 Aug 2009 14:30:34 +0200
Subject: [PATCH] set a soname

---
 libmultipath/Makefile |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/libmultipath/Makefile b/libmultipath/Makefile
index d59e2f3..3f2a48d 100644
--- a/libmultipath/Makefile
+++ b/libmultipath/Makefile
@@ -4,7 +4,9 @@
 #
 include ../Makefile.inc
 
-LIBS = libmultipath.so
+SONAME=0d
+DEVLIB = libmultipath.so
+LIBS = $(DEVLIB).$(SONAME)
 
 OBJS = memory.o parser.o vector.o devmapper.o callout.o \
        hwtable.o blacklist.o util.o dmparser.o config.o \
@@ -23,7 +25,8 @@ endif
 all: $(LIBS)
 
 $(LIBS): $(OBJS)
-	$(CC) $(SHARED_FLAGS) -Wl,-soname,$@ $(CFLAGS) -o $@ $(OBJS)
+	$(CC) $(SHARED_FLAGS) -Wl,-soname=$@ $(CFLAGS) -o $@ $(OBJS)
+	ln -sf $@ $(DEVLIB)
 
 install:
 	$(INSTALL_PROGRAM) -d $(DESTDIR)$(syslibdir)
@@ -34,4 +37,4 @@ uninstall:
 	rm -f $(DESTDIR)$(syslibdir)/$(LIBS)
 
 clean:
-	rm -f core *.a *.o *.gz *.so
+	rm -f core *.a *.o *.gz *.so *.so.*
-- 
