From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 30 Aug 2009 14:18:21 +0200
Subject: [PATCH] add library dependencies

---
 libmultipath/Makefile |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/libmultipath/Makefile b/libmultipath/Makefile
index 3f2a48d..ec1ab9f 100644
--- a/libmultipath/Makefile
+++ b/libmultipath/Makefile
@@ -7,6 +7,7 @@ include ../Makefile.inc
 SONAME=0d
 DEVLIB = libmultipath.so
 LIBS = $(DEVLIB).$(SONAME)
+LIBDEPS = -lpthread -ldl -ldevmapper
 
 OBJS = memory.o parser.o vector.o devmapper.o callout.o \
        hwtable.o blacklist.o util.o dmparser.o config.o \
@@ -25,7 +26,7 @@ endif
 all: $(LIBS)
 
 $(LIBS): $(OBJS)
-	$(CC) $(SHARED_FLAGS) -Wl,-soname=$@ $(CFLAGS) -o $@ $(OBJS)
+	$(CC) $(SHARED_FLAGS) $(LIBDEPS) -Wl,-soname=$@ $(CFLAGS) -o $@ $(OBJS)
 	ln -sf $@ $(DEVLIB)
 
 install:
-- 
